src/OpenLoco/Windows/CompanyWindow.cpp

```diff
@@ -2271,16 +2277,87 @@ namespace OpenLoco::Ui::Windows::CompanyWindow
             Common::repositionTabs(self);
         }
 
+        // this will prepare _commonFormatArgs array before drawing the StringIds::challenge_value
+        // after that for example it will draw this string: Achieve a performance index of 10.0% ("Engineer")
+        // 0x004384E9
+        static void sub_4384E9()
+        {
+            call(0x004384E9);
+        }
+
         // 0x00433DEB
         static void draw(window* self, Gfx::drawpixelinfo_t* dpi)
         {
             self->draw(dpi);
             Common::drawTabs(self, dpi);
 
-            registers regs;
-            regs.edi = (int32_t)dpi;
-            regs.esi = (int32_t)self;
-            call(0x00433DF5, regs);
+            char* buffer_2039 = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
+            *buffer_2039++ = static_cast<char>(ControlCodes::colour_black);
+            char* scenarioTitle = &_526114[0];
+            strcpy(buffer_2039, scenarioTitle);
+
+            int16_t y = self->y + 47;
+            // for example: "Provide the transport services on this little island" for "Boulder Breakers" scenario
+            y = Gfx::drawString_495224(*dpi, self->x + 5, y, self->width - 10, 0, StringIds::buffer_2039) + 5;
+            Gfx::drawString_494B3F(*dpi, self->x + 5, y, 0, StringIds::challenge_label);
+            y += 10;
+
+            FormatArguments args{};
+            sub_4384E9();
+            y = Gfx::drawString_495224(*dpi, self->x + 5, y, self->width - 10, 0, StringIds::challenge_value, &args) + 5;
+
+            company* playerCompany = CompanyManager::getPlayerCompany();
+
+            if ((playerCompany->challenge_flags & challenge_completed) != 0)
+            {
+                uint16_t years = _526245 / 12;
+                uint16_t months = _526245 % 12;
+
+                args = {};
+                args.push(years);
+                args.push(months);
+                Gfx::drawString_495224(*dpi, self->x + 5, y, self->width - 10, 0, StringIds::success_you_completed_the_challenge_in_years_months, &args);
+                return;
+            }
+
+            if ((playerCompany->challenge_flags & challenge_failed) != 0)
+            {
+                Gfx::drawString_495224(*dpi, self->x + 5, y, self->width - 10, 0, StringIds::failed_you_failed_to_complete_the_challenge);
+                return;
+            }
+
+            if ((playerCompany->challenge_flags & challenge_beaten_by_opponent) != 0)
+            {
+                uint16_t years = _526245 / 12;
+                uint16_t months = _526245 % 12;
+
+                args = {};
```

Comment:
In general I like to create a whole new args rather than reinitialise it again. If you wrapped the first use within some extra {} then you can reuse the name as well.